### PR TITLE
refactor(releases): refactor GetReleases for easier unit testing

### PR DESCRIPTION
**Pull request by jpower432**
# Description

This PR changes the way releases are queued up and processed to make unit testing easier. This allows for testing of what will be downloaded without actually having to contact an external API.

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] Unit tests added to ensure downloads are being created appropriately

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by afflom** [comment]
I've tested this by causing calculation forward and backward across minor version channels. The calculations behaved as expected under all conditions. lgtm. @estroz please merge when ready.

File: pkg/cincinnati/cincinnati.go [hunk 1]
```diff
@@ -165,14 +165,15 @@ func (c Client) GetUpdates(ctx context.Context, uri *url.URL, arch string, chann
 	return current, requested, updates, nil
 }
 
-func (c Client) CalculateUpgrades(ctx context.Context, uri *url.URL, arch, sourceChannel, targetChannel string, version semver.Version, reqVer semver.Version) (Update, Update, []Update, error) {
-	var requested Update
+// CalculateUpgrades fetches and calculates all the update payloads from the specified
+// upstream Cincinnati stack given the current and target version and channel
+func (c Client) CalculateUpgrades(ctx context.Context, uri *url.URL, arch, sourceChannel, targetChannel string, version semver.Version, reqVer semver.Version) (current Update, requested Update, upgrades []Update, err error) {
```
Comment (estroz) [inline pkg/cincinnati/cincinnati.go:170]: To make the signature smaller
```suggestion
func (c Client) CalculateUpgrades(ctx context.Context, uri *url.URL, arch, sourceChannel, targetChannel string, version, reqVer semver.Version) (current, requested Update, upgrades []Update, err error) {
```

File: pkg/cincinnati/find.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ func FindLastRelease(meta v1alpha1.Metadata, channel, url string, uuid uuid.UUID
 	return keys[len(keys)-1], vers[keys[len(keys)-1]], nil
 }
 
-func findLastReleases(meta v1alpha1.Metadata, url string, uuid uuid.UUID) (map[string]semver.Version, error) {
+func findLastReleases(meta v1alpha1.Metadata, uuid uuid.UUID) (map[string]semver.Version, error) {
```
Comment (estroz) [inline pkg/cincinnati/find.go:33]: `uuid` isn't used here either.

**Pull request merged**